lib/backup/s3remote/s3.go

```diff
@@ -178,6 +190,24 @@ func (fs *FS) Init(ctx context.Context) error {
 		// We manage upload concurrency by ourselves.
 		u.Concurrency = 1
 	})
+
+	m := make(map[string]*string)
+	for k, v := range fs.Metadata {
+		m[k] = &v
+	}
+	fs.metadata = m
+
+	if len(fs.Tags) > 0 {
+		t := ""
+		for k, v := range fs.Tags {
```

Comment:
Should tags be sorted to produce a stable value for tags string?

Reply:
Added sorting in https://github.com/VictoriaMetrics/VictoriaMetrics/pull/8995/commits/f31001cc886ca055329631c64a4a371aaebb15db.